amoro-ams/amoro-ams-server/src/main/java/org/apache/amoro/server/AmsServiceMetrics.java

```diff
@@ -0,0 +1,160 @@
+  private final MetricRegistry registry;
+  private List<MetricKey> registeredMetricKeys = Lists.newArrayList();
+
+  public AmsServiceMetrics(MetricRegistry registry) {
+    this.registry = registry;
+  }
+
+  public void register() {
+    registerHeapMetric();
+    registerThreadMetric();
+    registerCPuMetric();
+    registerGarbageCollectorMetrics();
+  }
+
+  public void unregister() {
+    registeredMetricKeys.forEach(registry::unregister);
+    registeredMetricKeys.clear();
+  }
+
+  private void registerHeapMetric() {
+    MemoryUsage heapMemoryUsage = ManagementFactory.getMemoryMXBean().getHeapMemoryUsage();
+    registerMetric(
+        registry,
+        AMS_STATUS_JVM_MEMORY_HEAP_USED,
+        (org.apache.amoro.api.metrics.Gauge<Long>) () -> heapMemoryUsage.getUsed());
+
+    registerMetric(
+        registry,
+        AMS_STATUS_JVM_MEMORY_HEAP_COMMITTED,
+        (org.apache.amoro.api.metrics.Gauge<Long>) () -> heapMemoryUsage.getCommitted());
+
+    registerMetric(
+        registry,
+        AMS_STATUS_JVM_MEMORY_HEAP_MAX,
+        (org.apache.amoro.api.metrics.Gauge<Long>) () -> heapMemoryUsage.getMax());
+  }
+
+  private void registerThreadMetric() {
+    registerMetric(
+        registry,
+        AMS_STATUS_JVM_THREADS_COUNT,
+        (org.apache.amoro.api.metrics.Gauge<Integer>)
+            () -> ManagementFactory.getThreadMXBean().getThreadCount());
+  }
+
+  private void registerCPuMetric() {
+    final com.sun.management.OperatingSystemMXBean mxBean =
+        (com.sun.management.OperatingSystemMXBean) ManagementFactory.getOperatingSystemMXBean();
+    registerMetric(
+        registry,
+        AMS_STATUS_JVM_CPU_LOAD,
+        (org.apache.amoro.api.metrics.Gauge<Double>) () -> mxBean.getProcessCpuLoad());
+    registerMetric(
+        registry,
+        AMS_STATUS_JVM_CPU_TIME,
+        (org.apache.amoro.api.metrics.Gauge<Long>) () -> mxBean.getProcessCpuTime());
+  }
+
+  private void registerGarbageCollectorMetrics() {
+    List<GarbageCollectorMXBean> garbageCollectorMXBeans =
+        ManagementFactory.getGarbageCollectorMXBeans();
+
+    for (final GarbageCollectorMXBean garbageCollector : garbageCollectorMXBeans) {
+      MetricDefine count = formatGarbageCollectorMetricDefine(garbageCollector.getName(), "count");
+      MetricDefine time = formatGarbageCollectorMetricDefine(garbageCollector.getName(), "time");
+      registerMetric(
+          registry,
+          count,
+          (org.apache.amoro.api.metrics.Gauge<Long>) () -> garbageCollector.getCollectionCount());
+      registerMetric(
+          registry,
+          time,
+          (org.apache.amoro.api.metrics.Gauge<Long>) () -> garbageCollector.getCollectionTime());
+    }
+  }
+
+  private MetricDefine formatGarbageCollectorMetricDefine(
+      String garbageCollector, String identify) {
+    return defineGauge(
+            AMS_STATUS_JVM_GARBAGE_COLLECTOR
+                + "_"
+                + garbageCollector.replaceAll("\\s", "_")
```

Comment:
Maybe we should define grabage collector name as a tag?

Reply:
good idea!

Reply:
> Maybe we should define grabage collector name as a tag?

good idea!